cmd/site-replication.go

```diff
@@ -3994,11 +3994,14 @@ func (c *SiteReplicationSys) healBucketReplicationConfig(ctx context.Context, ob
 		replMismatch = true
 	}
 
-	// validate remote targets on current cluster for this bucket
-	_, apiErr := validateReplicationDestination(ctx, bucket, rcfg, false)
-	if apiErr != noError {
-		replMismatch = true
+	if rcfg != nil {
+		// validate remote targets on current cluster for this bucket
+		_, apiErr := validateReplicationDestination(ctx, bucket, rcfg, false)
+		if apiErr != noError {
+			replMismatch = true
+		}
 	}
```

Comment:
added a validation further up to return error if rcfg is nil. The apiErr is being ignored here because the purpose is to heal replication config if it is using a stale ARN and to recreate targets and replication rules as needed for the cluster